chore!: noir contracts, notes and note abstractions cleanup

benesjan opened the pull request:
Fixes #15959
Partially addresses https://github.com/AztecProtocol/aztec-packages/issues/15966

There was a lot of mess lying around the codebase which had a real cost when it comes to maintenance. In this PR I:
- moved `BalanceSet` from token contract to a separate `balance_set` crate,
- drop the copies of `balance_set.nr` and use the one shared crate,
- drop `BalancesMap` from token blacklist and use there `BalanceSet`,
- drop `EasyPrivateUint` and use `BalanceSet` instead,
- drop the ugly utils from `ValueNote` and either use `BalanceSet` instead or I copy the utils to the test contracts where they were used (those test contracts are an ugly mess already so hiding it there seems fine),
- renamed `ValueNote` as `FieldNote`,
- dropped stale `DocsExample` contract and pointed readers to run `nargo expand` instead.

Now I can live in peace.

benesjan commented:
* **#18792** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18792?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18815** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18815?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18782** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18782?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18782?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#18721** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18721?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on playground/vite.config.ts:
```diff
@@ -141,9 +141,15 @@ export default defineConfig(({ mode }) => {
           description: 'Main entrypoint, hard limit',
         },
+        // Bump log:
+        // - Dec 2025: bumped from 4000 => 4500 as SimpleToken artifact grew to ~4485 KB causing CI build failures. This
+        // raise in artifact size was caused by the BalanceSet state variable being moved to a separate crate in this
+        // PR: https://github.com/AztecProtocol/aztec-packages/pull/18782.
+        // Not sure why this triggered raise in artifact size, but will not deal with this now as Grego's feedback is
+        // greatly needed here.
         {
           pattern: /.*/,
-          maxSizeKB: 4000,
+          maxSizeKB: 4500,
```
Sent a scheduled message to Grego to check this out once he gets back.

benesjan commented on docs/docs/developers/docs/aztec-nr/framework-description/functions/attributes.md:
```diff
@@ -21,53 +21,42 @@ A private function operates on private information, and is executed by the user
 
-To help illustrate how this interacts with the internals of Aztec and its kernel circuits, we can take an example private function, and explore what it looks like after Aztec.nr's macro expansion.
+If you are interested in what exactly the macros are doing we encourage you to run `nargo expand` on your contract.
+This will display your contract's code after the transformations are performed.
```
This can be seen as regression as now devs need to run `nargo expand` manually but I think it's the better tradeoff because they will get to see a fresh code and not the stale thing that was in `DocsExample` contract.

Also these docs are now very old and I am not sure if they'll even survive for long.

benesjan commented on docs/docs/developers/docs/aztec-nr/framework-description/functions/attributes.md:
```diff
@@ -21,53 +21,42 @@ A private function operates on private information, and is executed by the user
 
 The kernel checks that all of the values passed to each circuit in a function call are the same.
 
-**Returning the context to the kernel.**
-#include_code context-example-return /noir-projects/noir-contracts/contracts/docs/docs_example_contract/src/main.nr rust
-
-The contract function must return information about the execution back to the kernel. This is done through a rigid structure we call the `PrivateCircuitPublicInputs`.
-
-> _Why is it called the `PrivateCircuitPublicInputs`?_
-> When verifying zk programs, return values are not computed at verification runtime, rather expected return values are provided as inputs and checked for correctness. Hence, the return values are considered public inputs.
-
-This structure contains a host of information about the executed program. It will contain any newly created nullifiers, any messages to be sent to l2 and most importantly it will contain the return values of the function.
```
Moved this below as the ordering here was weird.

mverzilli reviewed (approved):
Nice torching session 🔥

benesjan committed:
chore!: noir contracts, notes and note abstractions cleanup

Fixes #15959
Partially addresses https://github.com/AztecProtocol/aztec-packages/issues/15966

There was a lot of mess lying around the codebase which had a real cost when it comes to maintenance. In this PR I:
- moved `BalanceSet` from token contract to a separate `balance_set` crate,
- drop the copies of `balance_set.nr` and use the one shared crate,
- drop `BalancesMap` from token blacklist and use there `BalanceSet`,
- drop `EasyPrivateUint` and use `BalanceSet` instead,
- drop the ugly utils from `ValueNote` and either use `BalanceSet` instead or I copy the utils to the test contracts where they were used (those test contracts are an ugly mess already so hiding it there seems fine),
- renamed `ValueNote` as `FieldNote`,
- dropped stale `DocsExample` contract and pointed readers to run `nargo expand` instead.

Now I can live in peace.
